infra/modules/notifications/resources/variables.tf

```diff
@@ -0,0 +1,20 @@
+variable "domain_name" {
+  description = "The domain name to configure SES."
+  type        = string
+}
+
+variable "name" {
+  type        = string
+  description = "Name of the notifications project/application"
+}
+
+variable "sender_email" {
+  type        = string
+  description = "Email address to use to send notification emails"
+}
```

Comment:
nice to have, can we add a [validation rule](https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules) that sender email is a string that ends with `"@${var.domain_name}"`

Reply:
unfortunately not possible:

```
│ Error: Invalid reference in variable validation
│
│   on ../../modules/notifications/resources/variables.tf line 16, in variable "sender_email":
│   16:     condition     = can(regex("^.+@${var.domain_name}$", var.sender_email))
│
│ The condition for variable "sender_email" can only refer to the variable itself, using var.sender_email.
```

Reply:
Looks like it's possible in Terraform 1.9.0
https://github.com/hashicorp/terraform/blob/v1.9/CHANGELOG.md#190-june-26-2024

I'm fine upgrading Terraform version now or punting this to follow up PR if you want. Defer to you. If you punt to a follow up PR can you create a ticket?

Reply:
Yeah I'll make a ticket for this one

Reply:
ticket: https://github.com/navapbc/template-infra/issues/777